server/test/write.go

```diff
@@ -733,6 +752,335 @@ var writeCommandTests = []writeCommandTest{
+			},
+			{
+				Type: "group",
+				Relations: map[string]*openfgapb.Userset{
+					"member": {Userset: &openfgapb.Userset_This{}},
+				},
+			},
+			{
+				Type: "document",
+				Relations: map[string]*openfgapb.Userset{
+					"reader": {Userset: &openfgapb.Userset_This{}},
+				},
+				Metadata: &openfgapb.Metadata{
+					Relations: map[string]*openfgapb.RelationMetadata{
+						"reader": {
+							DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+								{
+									Type:     "group",
+									Relation: "member",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		request: &openfgapb.WriteRequest{
+			Writes: &openfgapb.TupleKeys{TupleKeys: []*openfgapb.TupleKey{
+				tuple.NewTupleKey("document:budget", "reader", "group:abc#member"),
+			}},
+		},
+	},
+	{
+		_name: "Write succeeds if user is * and type references a specific type",
+		// state
+		schemaVersion: typesystem.SchemaVersion1_1,
+		typeDefinitions: []*openfgapb.TypeDefinition{
+			{
+				Type:      "user",
+				Relations: map[string]*openfgapb.Userset{},
+			},
+			{
+				Type: "group",
+				Relations: map[string]*openfgapb.Userset{
+					"member": {Userset: &openfgapb.Userset_This{}},
+				},
+			},
+			{
+				Type: "document",
+				Relations: map[string]*openfgapb.Userset{
+					"reader": {Userset: &openfgapb.Userset_This{}},
+				},
+				Metadata: &openfgapb.Metadata{
+					Relations: map[string]*openfgapb.RelationMetadata{
+						"reader": {
+							DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+								{
+									Type: "group",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		request: &openfgapb.WriteRequest{
+			Writes: &openfgapb.TupleKeys{TupleKeys: []*openfgapb.TupleKey{
+				tuple.NewTupleKey("document:budget", "reader", "*"),
+			}},
+		},
+	},
+	{
+		_name: "Write fails if user is * and type does not reference a specific type",
+		// state
+		schemaVersion: typesystem.SchemaVersion1_1,
+		typeDefinitions: []*openfgapb.TypeDefinition{
+			{
+				Type:      "user",
+				Relations: map[string]*openfgapb.Userset{},
+			},
+			{
+				Type: "group",
+				Relations: map[string]*openfgapb.Userset{
+					"member": {Userset: &openfgapb.Userset_This{}},
+				},
+			},
+			{
+				Type: "document",
+				Relations: map[string]*openfgapb.Userset{
+					"reader": {Userset: &openfgapb.Userset_This{}},
+				},
+				Metadata: &openfgapb.Metadata{
+					Relations: map[string]*openfgapb.RelationMetadata{
+						"reader": {
+							DirectlyRelatedUserTypes: []*openfgapb.RelationReference{
+								{
+									Type:     "group",
+									Relation: "member",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		request: &openfgapb.WriteRequest{
+			Writes: &openfgapb.TupleKeys{TupleKeys: []*openfgapb.TupleKey{
+				tuple.NewTupleKey("document:budget", "reader", "*"),
+			}},
+		},
+		err: serverErrors.InvalidTuple("User=* is not allowed to have relation reader with document:budget",
```

Comment:
I think it's probably better to just say "The wildcard '*' is an invalid user value for the 'document' relation 'reader'"

Reply:
Hmmm i'm hesitant to say "wildcard". https://docs.fga.dev/modeling/basics/public-access#02-check-that-the-relationship-exists

Reply:
Why are you hesitant to say it? We refer to it in that doc page multiple times.
<img width="912" alt="Screen Shot 2022-09-26 at 9 53 07 AM" src="https://user-images.githubusercontent.com/2899204/192323435-e0acd7d5-e623-4418-b0e4-219670005864.png">

Reply:
Because the doc says "it is not a wildcard" 😅

Reply:
But the heading says "Wildcard Syntax Usage" 🤣

Reply:
Following up this discussion: https://auth0.slack.com/archives/C033LQD4S11/p1664223016524409?thread_ts=1663955675.635259&cid=C033LQD4S11